vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -178,16 +178,22 @@ func validateVPA(vpa *vpa_types.VerticalPodAutoscaler) error {
 		}
 	}
 
+	if isCreate {
+		if vpa.Spec.TargetRef == nil {
+			return fmt.Errorf("TargetRef is required")
```

Comment:
Actually one more thing, can you make this more descriptive? Sth like, you are trying to use an unsupported version of the api, please migrate to v1beta2 and use targetRef to specify pods to autoscale

Reply:
Reworded.
Since we're not sure of the API version used here, it makes the message a bit long but maybe it's for the best.